cmd/snap/cmd_run.go

```diff
@@ -118,6 +118,40 @@ func getSnapInfo(snapName string, revision snap.Revision) (*snap.Info, error) {
 	return info, nil
 }
 
+func createOrUpdateUserDataSymlink(info *snap.Info, usr *user.User) error {
+	// create /home/snap/$SNAP_NAME/.lock file
+	lockFilePath := filepath.Join(info.HomeDirBase(usr.HomeDir), ".lock")
+	lockFd, err := syscall.Open(lockFilePath, syscall.O_CREAT, syscall.S_IRUSR|syscall.S_IWUSR)
```

Comment:
Using syscalls directly will bite us a bit. I wonder if there's a thin golang wrapper around them that still expresses the same low-level concepts?